app/validators/status_validator.rb

```diff
@@ -0,0 +1,9 @@
+class StatusValidator < ActiveModel::Validator
+  def validate(record)
+    return unless record.status_changed?
+    previous_status = record.status_was
+    record.errors.add(:status, 'invalid') if previous_status == 'open' && record.status != 'pending'
+    record.errors.add(:status, 'invalid') if previous_status == 'pending' && record.status != 'delivered'
+    record.errors.add(:status, 'invalid') if previous_status == 'delivered'
```

Comment:
just as a side note, for the future it would be better to use `state_machine` here :-)